paimon-format/src/main/java/org/apache/paimon/format/parquet/ParquetReaderFactory.java

```diff
@@ -155,13 +156,37 @@ private MessageType clipParquetSchema(GroupType parquetSchema) {
                         ParquetSchemaConverter.convertToParquetType(fieldName, projectedTypes[i]);
                 unknownFieldsIndices.add(i);
             } else {
-                types[i] = parquetSchema.getType(fieldName);
+                Type type = parquetSchema.getType(fieldName);
+                if (type instanceof GroupType && projectedTypes[i] instanceof RowType) {
+                    type = clipParquetGroup((GroupType) type, (RowType) projectedTypes[i]);
+                }
+                types[i] = type;
             }
         }
 
         return Types.buildMessage().addFields(types).named("paimon-parquet");
     }
 
+    /** Clips `parquetGroup` by `requestedRowType`. */
+    private GroupType clipParquetGroup(GroupType parquetGroup, RowType requestedRowType) {
+        Types.GroupBuilder<GroupType> builder = Types.buildGroup(parquetGroup.getRepetition());
+        List<DataField> fields = requestedRowType.getFields();
+        for (DataField field : fields) {
+            String fieldName = field.name();
+            if (parquetGroup.containsField(fieldName)) {
+                Type type = parquetGroup.getType(fieldName);
+                if (type instanceof GroupType && field.type() instanceof RowType) {
```

Comment:
Can Spark push down nested fields for array and map?

Reply:
Not support, only support  struct col pruning

![image](https://github.com/user-attachments/assets/28c54d70-4246-4361-a446-067452a8d054)

![image](https://github.com/user-attachments/assets/487ab9ac-355f-49b8-8ec9-5e0f260c1d9b)

Reply:
It support, update